### ISSUE TITLE
Can't read dumped BIOS from memory card

### ISSUE THREAD

**Issue by signer-ink-beast** [ShendoXT/biosdumper#3]
I actually have a memory card that gives me the same issues discussed in [this thread](https://www.ngemu.com/threads/cannot-read-dumped-psx-bios-data-from-the-memory-card-s.156047/) on the ngemu forum in 2013. This probably isn't the same card this person is talking about, but it gives me the same symptoms. Format card with my fat PS2, make sure its blank, plug it into slot 1 of the PSone I want to dump the BIOS of. It dumps successfully. I plug it back into the PS2, and... wLaunchElf shows the card is completely empty. The PS2 Browser says its unformatted. Put it back in the PSone without formatting it beforehand with the dumper already booted up, and the dumper also says its unformatted.

The memory card in question is a generic "RGEEK High Speed 1 Mega PS1 Memory Card". I purchased it on [Amazon](https://www.amazon.com/dp/B095C45TJD) recently.

I have done some testing on a PSone SCPH-101, a fat PS2 SCPH-30001R, and a slim PS2 (or PStwo) SCPH-90001. I wanted to use this card I got so that I can make use of tonyhax + FreePSXBoot on the other PS1 memory card I have in my possession: another generic, red card with an SJ logo on it that is probably an actual 1Mb-sized memory card (not tested, but it does work for the FreePSXBoot version of tonyhax).

I got the Constructor game and booted it up on my PSone. I start a game, press the select button and save. The game is supposed to use a whole, standard-sized 15-block (1M) memory card. And conveniently enough, the game counts the blocks it is in the progress of saving. What was interesting to me is that with this RGEEK card, it only gets up to 9. It appears to hang a bit, then jumps to verifying the save. Super handy for testing, right? But funnily enough, it claims success. I quit the current game to go back to the main menu so that I can load the save I just made. The game lists the save I just made. I select it, and after a couple of seconds it tells me the card is unformatted.

I repeat these steps over and over on my PSone and slim PS2. Sometimes I reset after saving and check the memory card in the browser or memory card menu. PS2 says it's unformatted, game says it's unformatted on PSone. Fat PS2 says it's unformatted. I cannot boot up Constructor on the fat PS2.

My guess is that this is a card that only really has 8 blocks to work with, and not 15. Either all these RGEEK cards are like this, or I have a lemon. If anyone else has access to this card, can you test it out with Constructor and post your results?

**Comment by ShendoXT** [comment]
Yes. There have been an influx of fake 64K cards lately. [More info here](https://psx0.wordpress.com/2022/04/01/fake-ps1-memory-cards-the-great-aliexpress-amazon-ebay-swindle/)

**Comment by signer-ink-beast** [comment]
Ah, that explains a lot. I have since ordered some legit PS1 cards to use instead of this one, so I should be able to get cracking on successfully dumping that BIOS in the near future. Good to know it seems to actually be 7 workable blocks rather than 8. I am glad I have always backed up my PS1 and PS2 saves up often (and back up the back ups). Caveat emptor.

Would it be worth mentioning that problems like these ones likely mean your card is less than 15 blocks in the Readme? Thanks again for your wonderful tool, ShendoXT. It makes dumping the PS1 BIOS simple and easy.
